tests/Functional/EligibleCatalogPromotionsProcessorTest.php

```diff
@@ -57,15 +57,17 @@ public function it_provides_catalog_promotions_with_precision_to_seconds(): void
             new \DateTime('2021-10-12 00:00:02'),
         ];
 
-        $actualDateTimes = [];
-
-        /** @var CatalogPromotionInterface $eligibleCatalogPromotion */
-        foreach ($eligibleCatalogPromotions as $eligibleCatalogPromotion) {
-            $actualDateTimes[] = $eligibleCatalogPromotion->getStartDate();
+        $actualDateTimes = array_map(
+            function (CatalogPromotionInterface $eligibleCatalogPromotion) {
```

Comment:
```suggestion
            function (CatalogPromotionInterface $eligibleCatalogPromotion): \DateTime {
```
or even lets use arrow function